src/Operation.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders;
+
+interface Operation
+{
+    /**
+     * Transform the given set of headers
+     *
+     * @param HeaderBag $headers
+     * @return void
+     */
+    public function modify(HeaderBag $headers);
```

Comment:
No, not necessary. Objects are de-facto passed by reference. Technically, it is still passing by value, see the first comment on [this SO answer](http://stackoverflow.com/a/881/192741) for the more correct explanation.